plugins/extractors/tableau/tableau.go

```diff
@@ -0,0 +1,190 @@
+	Host     string `mapstructure:"host" validate:"required"`
+	Version  string `mapstructure:"version" validate:"required"` // float as string
+	Username string `mapstructure:"username" validate:"required"`
+	Password string `mapstructure:"password" validate:"required"`
+	Sitename string `mapstructure:"sitename"`
+}
+
+// Extractor manages the extraction of data
+// from tableau server
+type Extractor struct {
+	config     Config
+	logger     log.Logger
+	httpClient *http.Client
+	client     Client
+}
+
+// Option provides extension abstraction to Extractor constructor
+type Option func(*Extractor)
+
+// WithHTTPClient assign custom http client to the Extractor constructor
+func WithHTTPClient(hcl *http.Client) Option {
+	return func(e *Extractor) {
+		e.httpClient = hcl
+	}
+}
+
+// New returns pointer to an initialized Extractor Object
+func New(logger log.Logger, opts ...Option) *Extractor {
+	e := &Extractor{
+		logger: logger,
+	}
+
+	for _, opt := range opts {
+		opt(e)
+	}
+
+	e.client = NewClient(e.httpClient)
+	return e
+}
+
+// Info returns the brief information of the extractor
+func (e *Extractor) Info() plugins.Info {
+	return plugins.Info{
+		Description:  "Dashboard list from Tableau server",
+		SampleConfig: sampleConfig,
+		Summary:      summary,
+		Tags:         []string{"oss", "extractor"},
+	}
+}
+
+// Validate validates the configuration of the extractor
+func (e *Extractor) Validate(configMap map[string]interface{}) (err error) {
+	return utils.BuildConfig(configMap, &Config{})
+}
+
+func (e *Extractor) Init(ctx context.Context, configMap map[string]interface{}) (err error) {
+	// build and validate config
+	err = utils.BuildConfig(configMap, &e.config)
+	if err != nil {
+		return plugins.InvalidConfigError{}
+	}
+
+	err = e.client.Init(ctx, e.config)
+	if err != nil {
+		return
+	}
+
+	return nil
+}
+
+// Extract collects metadata from the source. The metadata is collected through the out channel
+func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error) {
+	projects, err := e.client.GetAllProjects(ctx)
+	if err != nil {
+		err = errors.Wrap(err, "failed to fetch list of projects")
+		return
+	}
+
+	for _, proj := range projects {
+		workbooks, errWb := e.client.GetDetailedWorkbooksByProjectName(ctx, proj.Name)
+		if err != nil {
+			err = errors.Wrap(errWb, "failed to fetch list of workbook")
+			return
+		}
+		for _, wb := range workbooks {
+			dashboard, err := e.buildDashboard(wb)
+			if err != nil {
+				e.logger.Error("failed to build dashboard data of %v with error %s", wb, err.Error())
+				continue
+			}
+			emit(models.NewRecord(dashboard))
+		}
+	}
+	return
+}
+
+func (e *Extractor) buildDashboard(wb *Workbook) (data *assets.Dashboard, err error) {
+
+	dashboardURN := models.DashboardURN("tableau", e.config.Host, fmt.Sprintf("workbook/%s", wb.ID))
+	data = &assets.Dashboard{
+		Resource: &common.Resource{
+			Urn:     dashboardURN,
+			Name:    wb.Name,
+			Service: "tableau",
```

Comment:
The latest update in `main` branch added `Description` field to `Dashboard` model.
We can put workbook description inside the field.

```suggestion
			Service: "tableau",
			Description: wb.Description,
```

Reply:
That is better! thanks stew, will change it.